<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>660</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <widget class="QLineEdit" name="macro_title">
   <property name="geometry">
    <rect>
     <x>150</x>
     <y>30</y>
     <width>300</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>11</pointsize>
    </font>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
   <property name="placeholderText">
    <string>Enter Macro Title</string>
   </property>
  </widget>
  <widget class="QTextEdit" name="macro_description">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>80</y>
     <width>560</width>
     <height>111</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="placeholderText">
    <string>Enter Macro Descriptions</string>
   </property>
  </widget>
  <widget class="QPushButton" name="btn_del_sheet">
   <property name="geometry">
    <rect>
     <x>110</x>
     <y>570</y>
     <width>30</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>9</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>-</string>
   </property>
   <property name="flat">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QPushButton" name="btn_add_sheet">
   <property name="geometry">
    <rect>
     <x>170</x>
     <y>570</y>
     <width>30</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>9</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>+</string>
   </property>
   <property name="flat">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QPushButton" name="btn_del_cell">
   <property name="geometry">
    <rect>
     <x>400</x>
     <y>570</y>
     <width>30</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>9</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="text">
    <string>-</string>
   </property>
   <property name="flat">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QPushButton" name="btn_add_cell">
   <property name="geometry">
    <rect>
     <x>460</x>
     <y>570</y>
     <width>30</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>9</pointsize>
     <weight>75</weight>
     <bold>true</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <property name="text">
    <string>+</string>
   </property>
   <property name="flat">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QPushButton" name="btn_cancel">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>620</y>
     <width>91</width>
     <height>23</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Cacel</string>
   </property>
  </widget>
  <widget class="QPushButton" name="btn_save">
   <property name="geometry">
    <rect>
     <x>330</x>
     <y>620</y>
     <width>91</width>
     <height>23</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Arial</family>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </widget>
  <widget class="QWidget" name="">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>220</y>
     <width>520</width>
     <height>331</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QTableWidget" name="macro_sheetview">
      <property name="selectionMode">
       <enum>QAbstractItemView::SingleSelection</enum>
      </property>
      <property name="columnCount">
       <number>2</number>
      </property>
      <attribute name="horizontalHeaderVisible">
       <bool>true</bool>
      </attribute>
      <attribute name="horizontalHeaderCascadingSectionResizes">
       <bool>false</bool>
      </attribute>
      <attribute name="horizontalHeaderMinimumSectionSize">
       <number>50</number>
      </attribute>
      <attribute name="horizontalHeaderDefaultSectionSize">
       <number>30</number>
      </attribute>
      <attribute name="horizontalHeaderHighlightSections">
       <bool>false</bool>
      </attribute>
      <attribute name="horizontalHeaderStretchLastSection">
       <bool>true</bool>
      </attribute>
      <column/>
      <column/>
     </widget>
    </item>
    <item>
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>60</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="QTableWidget" name="macro_cellview">
      <property name="columnCount">
       <number>2</number>
      </property>
      <attribute name="horizontalHeaderMinimumSectionSize">
       <number>50</number>
      </attribute>
      <attribute name="horizontalHeaderDefaultSectionSize">
       <number>30</number>
      </attribute>
      <attribute name="horizontalHeaderHighlightSections">
       <bool>false</bool>
      </attribute>
      <attribute name="horizontalHeaderStretchLastSection">
       <bool>true</bool>
      </attribute>
      <column/>
      <column/>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <tabstops>
  <tabstop>macro_title</tabstop>
  <tabstop>macro_description</tabstop>
  <tabstop>macro_sheetview</tabstop>
  <tabstop>macro_cellview</tabstop>
  <tabstop>btn_del_sheet</tabstop>
  <tabstop>btn_add_sheet</tabstop>
  <tabstop>btn_del_cell</tabstop>
  <tabstop>btn_add_cell</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>macro_sheetview</sender>
   <signal>itemClicked(QTableWidgetItem*)</signal>
   <receiver>Dialog</receiver>
   <slot>select_sheet_item(QTableWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>186</x>
     <y>297</y>
    </hint>
    <hint type="destinationlabel">
     <x>-112</x>
     <y>777</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>macro_cellview</sender>
   <signal>itemClicked(QTableWidgetItem*)</signal>
   <receiver>Dialog</receiver>
   <slot>select_cell_item(QTableWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>402</x>
     <y>325</y>
    </hint>
    <hint type="destinationlabel">
     <x>-70</x>
     <y>790</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_del_sheet</sender>
   <signal>released()</signal>
   <receiver>Dialog</receiver>
   <slot>delete_sheet_item()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>125</x>
     <y>581</y>
    </hint>
    <hint type="destinationlabel">
     <x>-26</x>
     <y>805</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_add_sheet</sender>
   <signal>released()</signal>
   <receiver>Dialog</receiver>
   <slot>add_sheet_item()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>173</x>
     <y>577</y>
    </hint>
    <hint type="destinationlabel">
     <x>70</x>
     <y>809</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_del_cell</sender>
   <signal>released()</signal>
   <receiver>Dialog</receiver>
   <slot>delete_cell_item()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>422</x>
     <y>582</y>
    </hint>
    <hint type="destinationlabel">
     <x>84</x>
     <y>820</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_add_cell</sender>
   <signal>released()</signal>
   <receiver>Dialog</receiver>
   <slot>add_cell_item()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>482</x>
     <y>580</y>
    </hint>
    <hint type="destinationlabel">
     <x>-34</x>
     <y>845</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_cancel</sender>
   <signal>released()</signal>
   <receiver>Dialog</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>207</x>
     <y>634</y>
    </hint>
    <hint type="destinationlabel">
     <x>-125</x>
     <y>849</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_save</sender>
   <signal>released()</signal>
   <receiver>Dialog</receiver>
   <slot>save_macro()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>379</x>
     <y>636</y>
    </hint>
    <hint type="destinationlabel">
     <x>292</x>
     <y>837</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>select_sheet_item(QTableWidgetItem*)</slot>
  <slot>select_cell_item(QTableWidgetItem*)</slot>
  <slot>delete_sheet_item()</slot>
  <slot>add_sheet_item()</slot>
  <slot>add_cell_item()</slot>
  <slot>delete_cell_item()</slot>
  <slot>save_macro()</slot>
  <slot>sheet_changed()</slot>
 </slots>
</ui>
